Tasks/AzureCloudPowerShellDeployment/Utility.ps1

```diff
@@ -109,7 +121,18 @@ function Get-DiagnosticsExtensions($storageAccount, $extensionsPath, $storageAcc
                         {
                             try
                             {
-                                $publicConfigStorageKey = Get-AzureStorageKey -StorageAccountName $publicConfigStorageAccountName
+                                $publicConfigStorageKey = Get-AzureStorageKey -StorageAccountName $publicConfigStorageAccountName -ErrorAction Ignore
+                                if ($publicConfigStorageKey -eq $null)
+                                {
+                                    #if the storage account is not the classic one, get the primary key in another way
+                                    $publicConfigStorageKey = (Get-AzureRmStorageAccountKey -ResourceGroupName "$storageAccountResourceGroup" -Name "$publicConfigStorageAccountName").Value[0]
```

Comment:
There are few tasks which works with both endpoint. In such tasks, we take connection type as input. Connection type can be _Azure Classic_ or _ARM_.  After connection type is being selected complete execution will be corresponding to selected endpoint either classic or ARM.

_Let me explain why this behavior is for Azure PS_ - 
In AzurePS task we support cmdlet corresponding to classic endpoint and ARM endpoint. So, you can execute only azure classic cmdlet if classic endpoint is being selected. Similarly, if ARM endpoint is being selected you can use only ARM cmdlets.

But Azure cloud service itself is a classic resource. So, there will be no ARM cmdlet corresponding to it. So ARM endpoint could be used here.